iphone/Classes/PlatformModule.m

```diff
@@ -297,17 +297,17 @@ - (NSNumber *)totalMemory
 }
 GETTER_IMPL(NSNumber *, availableMemory, AvailableMemory);
 
-- (BOOL)openURL:(NSString *)url withOptions:(id)options andCallback:(JSValue *)callback
+- (BOOL)openURL:(NSString *)url withOptions:(JSValue *)options andCallback:(JSValue *)callback
 {
   NSURL *newUrl = [TiUtils toURL:url proxy:self];
   BOOL result = NO;
 
-  // iOS 10+
   NSMutableDictionary *optionsDict = [NSMutableDictionary dictionary];
-  if ([options isKindOfClass:[NSDictionary class]]) {
-    optionsDict = (NSMutableDictionary *)options;
-  } else if ([options isKindOfClass:[JSValue class]]) {
-    callback = (JSValue *)options;
+  NSDictionary *dictionary = [options toDictionary];
+  if (dictionary.allKeys.count > 0) {
+    optionsDict = (NSMutableDictionary *)dictionary;
```

Comment:
I know it was like this before, but since you are already touching this piece of code: Casting to `NSMutableDictionary` is unnecessary here. Should someone ever try to write to the `optionsDict` here it will crash, since the underlying type is still `NSDictionary` (the proper way would have been using `mutableCopy`). `optionsDict`  never gets mutated anyway, so it should be a `NSDictionary`.